static/js/directives/chat.js

```diff
@@ -336,6 +340,19 @@ define(['jquery', 'underscore', 'text!partials/chat.html', 'text!partials/chatro
 						subscope.doClear = function() {
 							subscope.$broadcast("clear");
 						};
+						subscope.addContact = function() {
+							subscope.$emit("requestcontact", subscope.id, {
+								restore: true
+							});
+						};
+						subscope.removeContact = function() {
+							contacts.remove(buddy.contact.Userid);
+						};
+						subscope.updateContactStatus = function(event, data) {
+							subscope.isContact = event.type === "contactadded";
+						};
+						contacts.e.on("contactadded", subscope.updateContactStatus);
+						contacts.e.on("contactremoved", subscope.updateContactStatus);
```

Comment:
This does not make sense and needs fixing. This events trigger for all contacts and need to be limited/matched to the user in this subscope what so ever. In general i would like it if this two events could be avoided alltogether.